distributed/scheduler.py

```diff
@@ -5510,6 +5511,41 @@ def send_all(self, client_msgs: dict, worker_msgs: dict):
             except (CommClosedError, AttributeError):
                 self.loop.add_callback(self.remove_worker, address=worker)
 
+    def annotate_task(
+        self,
+        worker=None,
+        key: str = None,
+        annotations: Mapping = None,
+        resource_restrictions: Mapping = None,
```

Comment:
I recommend making restrictions a separate route.  I don't think that there is any reason for annotations and restrictions to be intertwined.

However, I do think that it would be useful to support a variety of different kinds of restrictions.  You should probably roll this into the `Scheduler.set_restrictions` method.  I think that it has a `worker=` keyword today as a hint that `host=` and `resource=` should come later.

Reply:
Good point. 
A related question, what is the different between regular handlers and stream handlers?
Can I use `self.batched_stream.send` for a `"op": "set_restrictions"` message when `set_restrictions` is in `Schduler.handlers`?

Reply:
Stream handlers are used with BatchedSend objects.  They are strictly fire and forget.  They're also very cheap to send.  We accumulate several small messages in a list and then send off that list periodically (with a very short period)

The handlers are what receive the `await rpc.foo(...)` commands.  They wait for and collect a response.  

If you need a response then use `await rpc.foo(...)`.  If you want to send lots of little small messages without really affecting overhead then try to use a stream handler.  You can certainly put a method in both stream handlers and handlers, although if the receiving end needs a response then this might require some cleverness.

Does that provide enough information?

Reply:
Yes, thanks!